### PR TITLE
layers: Add platform guards to enum_string_helper

### PR DESCRIPTION
**Pull request by charles-lunarg**
Now that validation layers generates source with BETA_EXTENSIONS, the enum_string_helper
must guard the enums from extensions in the beta header.

### CODE REVIEW

**Review by ncesario-lunarg — APPROVED** [review]
LGTM, thanks @charles-lunarg!

**Review by mikes-lunarg — APPROVED** [review]
LGTM. I tested locally with and without `VK_ENABLE_BETA_EXTENSIONS` and both worked.

File: scripts/helper_file_generator.py [hunk 1]
```diff
@@ -565,8 +568,12 @@ def GenerateEnumStringConversion(self, groupName, value_list, bitwidth):
         # Emit these in a repeatable order so file is generated with the same contents each time.
         # This helps compiler caching systems like ccache.
         for item in sorted(value_list):
+            if item in protect_dict:
```
Comment (mikes-lunarg) [inline scripts/helper_file_generator.py:571]: Need to add `protect_dict` to the `GenerateEnumStringConversion` arg list above

Comment (charles-lunarg) [reply scripts/helper_file_generator.py:571]: Should be fixed now, force pushed and running CI.

**Pull request merged**